chutney/server/src/main/java/com/chutneytesting/dataset/api/DataSetController.java

```diff
@@ -60,7 +60,11 @@ public DataSetDto save(@RequestBody DataSetDto datasetDto) {
     @PutMapping(path = "", produces = MediaType.APPLICATION_JSON_VALUE)
     public DataSetDto update(@RequestBody DataSetDto dataSetDto, @RequestParam Optional<String> oldId) {
         hasNoDuplicatedHeaders(dataSetDto);
-        return toDto(datasetService.update(oldId, fromDto(dataSetDto)));
+        if (oldId.isPresent()) {
+            return toDto(datasetService.update(oldId.get(), fromDto(dataSetDto)));
```

Comment:
maybe rename the method update with updateDatasetName() as it's now used only for that